alkira/resource_alkira_service_f5_lb_helper.go

```diff
@@ -0,0 +1,258 @@
+package alkira
+
+import (
+	"errors"
+	"log"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// expandF5Instances converts the input data to a slice of F5Instances structs.
+func expandF5Instances(in []interface{}, m interface{}) ([]alkira.F5Instance, error) {
+
+	client := m.(*alkira.AlkiraClient)
+
+	if in == nil || len(in) == 0 {
+		log.Printf("[ERROR] Invalid F5 Load Balancer instance input.")
+		return nil, errors.New("Invalid F5 Load Balancer instance input.")
+	}
+
+	instances := make([]alkira.F5Instance, len(in))
+	instanceDeployment := alkira.F5InstanceDeployment{}
+
+	for i, instance := range in {
+		tfInstance := instance.(map[string]interface{})
+		instanceStruct := alkira.F5Instance{}
+		if instanceId, ok := tfInstance["id"]; ok {
+
+			instanceStruct.Id = instanceId.(int)
+		}
+		if name, ok := tfInstance["name"]; ok {
+
+			instanceStruct.Name = name.(string)
+		}
+		if licenseType, ok := tfInstance["license_type"]; ok {
+			instanceStruct.LicenseType = licenseType.(string)
+			// if the license type is BRING_YOUR_OWN, we need a registration credential.
+			if licenseType == "BRING_YOUR_OWN" {
+				if regCredId, ok := tfInstance["registration_credential_id"].(string); ok {
```

Comment:
if you use `tfInstance["registration_credential_id"]` with `(string)`, it may crash if the former one is `null`, so it's better to cast it inside the `if`.